lib/rex/proto/kerberos/model/krb_cred_info.rb

```diff
@@ -132,6 +159,164 @@ def encode_srealm
+            # length is a multiple of 8 then _unused_bits_ is 0. Otherwise
+            # _unused_bits_ indicates the number of bits that are to be ignored in
+            # the final octet of the BitString's _value_.
+            unused_bits = input.value[0].unused_bits
+            flags >>= unused_bits
+            Rex::Proto::Kerberos::Model::KdcOptionFlags.new(flags)
+          end
+
+          # Decodes the auth_time field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Time]
+          def decode_auth_time(input)
+            input.value[0].value
+          end
+
+          # Decodes the start_time field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Time]
+          def decode_start_time(input)
+            input.value[0].value
+          end
+
+          # Decodes the end_time field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Time]
+          def decode_end_time(input)
+            input.value[0].value
+          end
+
+          # Decodes the renew_till field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Time]
+          def decode_renew_till(input)
+            input.value[0].value
+          end
+
+          # Decodes the srealm field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [String]
+          def decode_srealm(input)
+            input.value[0].value
+          end
+
+          # Decodes the prealm field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [String]
+          def decode_prealm(input)
+            input.value[0].value
+          end
+
+          # Decodes the sname field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Rex::Proto::Kerberos::Type::PrincipalName]
+          def decode_sname(input)
+            Rex::Proto::Kerberos::Model::PrincipalName.decode(input.value[0])
+          end
+
+          # Decodes the pname field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Rex::Proto::Kerberos::Type::PrincipalName]
+          def decode_pname(input)
+            Rex::Proto::Kerberos::Model::PrincipalName.decode(input.value[0])
+          end
+
+          # Decodes the caddr field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Array<Rex::Proto::Model::HostAddress>]
+          def decode_caddr(input)
+            caddr = []
+            input.value[0].value.each do |host_address_data|
+              caddr << Rex::Proto::Kerberos::Model::HostAddress.decode(host_address_data)
+            end
+            caddr
+          end
```

Comment:
This looks like this decodes the `caddr` field, but there's no attr_accessor defined for it, it's not handled in `#==`, `#encode` or `#decode_asn`.

Reply:
ah yea I see what happened here, I totally just copy pasted the decode implementation for `caddr` without realising it was never added into the model and based the encode, etc off of the `attr_accessor`s that were already a part of the model
Can add it in no problem assuming it wasn't left out initially on purpose